mono/metadata/object.c

```diff
@@ -3333,17 +3357,15 @@ mono_field_set_value (MonoObject *obj, MonoClassField *field, void *value)
  * The value must be in the native format of the field type. 
  */
 void
-mono_field_static_set_value (MonoVTable *vt, MonoClassField *field, void *value)
+mono_field_static_set_value_internal (MonoVTable *vt, MonoClassField *field, void *value)
 {
-	MONO_ENTER_GC_UNSAFE;
```

Comment:
Perhaps this PR should not have any GC mode change changes, i.e. do them in a separate PR.